numpy/core/src/umath/ufunc_object.c

```diff
@@ -2343,6 +2344,16 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
         }
     }
 
+    total_problem_size = NpyIter_GetIterSize(iter);
+    if (total_problem_size < 0) {
```

Comment:
Otherwise looks good to me.
.
Obligatory comment in passing that the code duplication between ufuncs and
gufuncs sucks.

On Fri, Feb 5, 2016 at 1:53 PM, seberg <notifications@github.com> wrote:

> In numpy/core/src/umath/ufunc_object.c
> <https://github.com/numpy/numpy/pull/7198#discussion_r52077858>:
>
> > @@ -2343,6 +2344,16 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
> >          }
> >      }
> >
> > +    total_problem_size = NpyIter_GetIterSize(iter);
> > +    if (total_problem_size < 0) {
>
> Actually I am not sure how reliable this is, but don't have a quick better
> idea either. With all those inner core dimensions, I guess the iterator
> size could be vastly overestimated possibly (I am not sure).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/7198/files#r52077858>.
>



-- 
Nathaniel J. Smith -- https://vorpus.org <http://vorpus.org>